dali/python/nvidia/dali/external_source.py

```diff
@@ -365,6 +365,14 @@ class ExternalSource():
 
     This argument will be required starting from DALI 2.0.
 
+`ndim` : int, optional
+    Input dimensionality.
+
+    The dimensionality of the data provided to the operator will be verified against this value.
+    Number of dimensions can be also inferred from the ``layout`` argument if provided.
+
```

Comment:
I'd also include here a sentence that if both are provided, they must match.

Reply:
done